### PR TITLE
Port time constant (de)convolution

### PR DESCRIPTION
**Pull request by keskitalo**

### CODE REVIEW

File: src/toast/tests/ops_time_constant.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+        # Simulate noise from this model
+        sim_noise = ops.SimNoise(noise_model="noise_model", out="signal")
+        sim_noise.apply(data)
+
+        # Copy the signal for reference
+        for obs in data.obs:
+            obs.detdata.ensure("signal0")
+            for det in obs.local_detectors:
+                obs.detdata["signal0"][det][:] = obs.detdata["signal"][det]
```
Comment (tskisner) [inline src/toast/tests/ops_time_constant.py:53]: This is fine, but for reference another way would be:
```
copy = ops.Copy(detdata=[("signal", "signal0")])
copy.apply(data)
```

File: workflows/toast_sim_ground.py [hunk 2]
```diff
@@ -325,15 +338,21 @@ def main():
             name="det_pointing_radec", quats="quats_radec"
         ),
         toast.ops.ScanHealpix(name="scan_map", enabled=False),
-        toast.ops.SimNoise(name="sim_noise"),
         toast.ops.SimAtmosphere(name="sim_atmosphere"),
+        toast.ops.TimeConstant(
```
Comment (tskisner) [inline workflows/toast_sim_ground.py:342]: Should we add this to the example satellite workflow as well?

**Review by tskisner — APPROVED** [review]
Thanks for adding this to our tool box!

File: src/toast/ops/time_constant.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            for det in dets:
+                signal = obs.detdata[self.det_data][det]
+
+                tau = self._get_tau(obs, det)
+                if self.deconvolve:
+                    taufilter = (1 + 2.0j * np.pi * freqs * tau.to_value(u.s))
+                else:
+                    taufilter = 1.0 / (1 + 2.0j * np.pi * freqs * tau.to_value(u.s))
+
+                signal[:] = np.fft.irfft(taufilter * np.fft.rfft(signal), n=nsample)
```
[the remark this answers is not in the document]
Comment (keskitalo) [reply src/toast/ops/time_constant.py:109]: Fair point. I added a comment to the source code about this.

File: src/toast/tests/ops_time_constant.py (hunk 1 above)
Comment (keskitalo) [reply src/toast/tests/ops_time_constant.py:53]: That is much cleaner! I changed it.

File: workflows/toast_sim_ground.py (hunk 2 above)
Comment (keskitalo) [reply workflows/toast_sim_ground.py:342]: Excellent point. Done!

**Pull request merged**